src/Generator.h

```diff
@@ -3332,6 +3376,20 @@ class GeneratorBase : public NamesInterface, public GeneratorContext {
         return p;
     }
 
+    // Create Output<Buffer> with compile-time type & dimensions
+    template<typename T,
+             typename std::enable_if<!std::is_arithmetic<T>::value && !std::is_same<T, Halide::Func>::value>::type * = nullptr>
+    GeneratorOutput<T> *add_output(const std::string &name) {
+        static_assert(T::has_static_dimensions, "You can only call this version of add_output() for a Buffer<T, D> where T is not void.");
```

Comment:
and here, unless I'm really not understanding something

Reply:
ditto